controller/engine_controller.go

```diff
@@ -1768,6 +1781,11 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 			// it is past the Backoff period, we'll try to mark the Replica as Failed and increase the Backoff period
 			// for the next failure.
 			if !ec.backoff.IsInBackOffSinceUpdate(e.Name, time.Now()) {
+				if err := ec.updateReplicaRebuildFailedCondition(replicaObj, replicaRebuildErrMsg); err != nil {
+					log.WithError(err).Errorf("Failed to update rebuild status information on replica %v", replica)
+					return
+				}
```

Comment:
The code below here is not required, because you already have queried `replica` already.

```go
rep, err := ec.ds.GetReplica(replica)
if err != nil {
`log.WithError(err).Errorf("Failed to get replica %v unable to mark failed rebuild", replica)
`return
}
```

Reply:
Removed. Thanks.